static/app/components/core/slider/index.chonk.tsx

```diff
@@ -0,0 +1,317 @@
+import {
+  type CSSProperties,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import type {DO_NOT_USE_ChonkTheme} from '@emotion/react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+import {mergeRefs} from '@react-aria/utils';
+
+import {space} from 'sentry/styles/space';
+
+import type {SliderProps} from './index';
+
+const passthrough = (n: number | '') => n;
+
+export function Slider({
+  formatLabel = passthrough,
+  ref,
+  ...props
+}: SliderProps & {ref: React.Ref<HTMLInputElement>}) {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const refs = mergeRefs(ref, inputRef);
+  const step = toNumber(props.step ?? -1);
+  const {value, min, max} = resolveMinMaxValue(props);
+  const [label, setLabel] = useState(value);
```

Comment:
Instead of storing state, how about just modify the label directly to formatLabel(value) and skip storing it?

Reply:
the component won’t re-render if we don’t store the label in `handleChange`, as this is done with a native event listener, but the label is rendered in react.

@natemoo-re why are we mixing the two? If we want the component to re-render when the slider changes (to update the label), why do we use a native event handler?

Reply:
Turns out the native `min` and `max` values report empty string when not set, which wasn’t handled. I fixed that here: c74eec3b

But it makes me want to move away from those native events towards having react handle this even more

Reply:
I refactored it towards using react `onChange`, please have another look @natemoo-re @JonasBa

Reply:
Awesome, thanks @TkDodo! I was using the native event in an attempt to allow uncontrolled usage as well as controlled, but it's probably much simpler to just force controlled usage.